ipatests/test_integration/test_caless.py

```diff
@@ -306,21 +303,53 @@ def get_replica_filename(self, replica):
                             'replica-info.gpg')
 
     @classmethod
-    def export_pkcs12(cls, nickname, filename='server.p12', password=None):
-        """Export a cert as PKCS#12 to the given file"""
+    def create_pkcs12(cls, nickname, filename='server.p12', password=None):
+        """Create a cert chain and generate pkcs12 cert"""
         if password is None:
             password = cls.cert_password
-        ipautil.run(['pk12util',
-                     '-o', filename,
-                     '-n', nickname,
-                     '-d', 'nssdb',
-                     '-K', cls.cert_password,
-                     '-W', password], cwd=cls.cert_dir)
+
+        fname_chain = []
+
+        key_fname = os.path.join(cls.cert_dir, nickname + '.key')
+        certchain_fname = os.path.join(cls.cert_dir, nickname + '.pem')
+
+        nick_chain = nickname.split(os.sep)
+
+        # to construct whole chain e.g "ca1 - ca/sub - ca1/sub/server"
+        for index, _value in enumerate(nick_chain):
+            cert_path = '/'.join(nick_chain[:index+1])
+            fname_chain.append(os.path.join(cls.cert_dir, cert_path + '.crt'))
+        # to simulate unknown CA we drop ca3 from the chain
+        if nickname.startswith('ca3/'):
```

Comment:
Please remove this and just append the files to `fname_chain` if they exist. `ca3.crt` is not generated as a part of `crreate_caless_pki` (or it was not originally, hopefully you did not change this).